core/eolearn/core/eodata_io.py

```diff
@@ -155,6 +156,136 @@ def walk_eopatch(eopatch, patch_location, features=...):
+                if feat not in eopatch_content.keys():
+                    eopatch_content[feat] = [eopatch[feat]]
+                else:
+                    if eopatch[feat].shape == eopatch_content[feat][-1].shape:
+                        if deep_eq(eopatch[feat], eopatch_content[feat][-1]) and timeless_op:
+                            raise ValueError(f'Two identical timeless arrays were found for {feat}.')
+                        eopatch_content[feat].append(eopatch[feat])
+                    else:
+                        raise ValueError(f'The arrays have mismatching n x m x b shapes for {feat}.')
+
+    duplicate_timestamp = eopatches[0].timestamp + [tstamp for eop, mask in zip(eopatches[1:], masks)
+                                                    for tstamp, to_keep in zip(eop.timestamp, mask) if not to_keep]
+    _check_duplicate_timestamps(eopatch_content, duplicate_timestamp)
+
+    sorted_indices = sorted(range(len(timestamp)), key=lambda k: timestamp[k])
+
+    eopatch = _perform_concat_operation(eopatches[0].__copy__(), eopatch_content_concat, sorted_indices)
+    eopatch = _perform_merge_operation(eopatch, eopatch_content, sorted_indices, time_dependent_op, timeless_op)
+
+    eopatch.timestamp = sorted(timestamp)
+
+    return eopatch
+
+
+def _check_allowed_operations(time_dependent_op, timeless_op):
+    """ Checks that no the passed operations for time-dependent and timeless feature merging are allowed.
+    """
+    allowed_timeless_op = [None, "mean", "max", "min", "median"]
+    allowed_time_dependent_op = ["concatenate", "mean", "max", "min", "median"]
+    if timeless_op not in allowed_timeless_op:
+        raise ValueError('timeless_op "%s" is invalid, must be one of %s' % (timeless_op, allowed_timeless_op))
+    if time_dependent_op not in allowed_time_dependent_op:
+        raise ValueError('time_dependent_op "%s" is invalid, must be one of %s'
+                         % (time_dependent_op, allowed_time_dependent_op))
+
+
+def _perform_concat_operation(eopatch, eopatch_content_concat, sorted_indices):
+    """ Performs the temporal concatenation of non-duplicate timestamps of time-dependent features.
+    """
+    for feat, arrays in eopatch_content_concat.items():
+        ftype, _ = feat
+        if ftype != FeatureType.VECTOR and sorted_indices:
+            eopatch[feat] = np.concatenate(arrays, axis=0)[sorted_indices]
+        else:
+            eopatch[feat] = pd.concat([array for array in arrays if not array.empty]).sort_values('TIMESTAMP')
+
+    return eopatch
+
+
+def _perform_merge_operation(eopatch, eopatch_content, sorted_indices, time_dependent_op, timeless_op):
+    """ Performs the merging of duplicate timestamps of time-dependent features and of timeless features.
+    """
+    for feat, arrays in eopatch_content.items():
+        ftype, _ = feat
+        if ftype.is_time_dependent() and ftype != FeatureType.VECTOR and sorted_indices:
+            if time_dependent_op == 'mean':
```

Comment:
To avoid all these if-else statements you can have a dictionary mapping operation strings to numpy functions: `{'mean': np.nanmean, ...}`.